pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -448,9 +455,92 @@ func (r *resourceManager) getRootCAVolumeSourceName(ctx context.Context) (string
 func (r *resourceManager) ensureDeployment(ctx context.Context) error {
 	deployment := r.emptyDeployment()
 
-	rootCAVolumeSourceName, err := r.getRootCAVolumeSourceName(ctx)
-	if err != nil {
-		return err
+	if version.ConstraintK8sLess124.Check(r.values.Version) {
+		var err error
+		rootCAVolumeSourceName, err = r.getRootCAVolumeSourceName(ctx)
+		if err != nil {
+			return err
+		}
+
+		volume = corev1.Volume{
```

Comment:
Instead of defining the entire volume, it should be sufficient to only define the respective source (`core1.VolumeProjection`). In this case it's a `Secret`, in the other case it's a `ConfigMap`. This way, we don't duplicate the common parts.